src/openrct2/ride/TrackDesign.cpp

```diff
@@ -2141,6 +2143,8 @@ void TrackDesignDrawPreview(TrackDesign& td, uint8_t* pixels)
         dpi.bits += kTrackPreviewImageSize;
     }
 
+    drawingEngine->EndDraw();
+
```

Comment:
We could likely be using RAII to make these neater with something like:

```c++
struct DrawingScope
{
    DrawingScope(auto* drawingEngine)
        : m_drawingEngine(drawingEngine)
    {
        m_drawingEngine->BeginDraw();
    }
    ~DrawingScope()
    {
        m_drawingEngine->EndDraw();
    }
    auto* drawingEngine;
}
```

Reply:
Perhaps, however I'm not sure its also worth it as this happens about 3 times. I'll consider it in the next cleansing PRs